A29-xds-tls-security.md

```diff
@@ -321,54 +347,38 @@ part of validating all LDS updates:
+plugin framework (described later). The field `instance_name` defines a CertificateProvider "instance" that gRPC looks
+up in the bootstrap file (described later) to obtain the CertificateProvider configuration. This configuration along
+with the CertificateProvider plugin framework enables gRPC to acquire the certificates necessary for the TLS handshake.
+
+The field [`tls_certificate_provider_instance`][TCPI] is used for the identity certificate and the
+private key. And [`ca_certificate_provider_instance`][CCPI] inside a `CertificateValidationContext`
+is used for the root certificates for validating peer certificates.
 
 gRPC does not support the other certificate acquisition mechanisms specified by the following fields:
 
-* [tls_certificates][TLS-CERT]
-* [tls_certificate_sds_secret_configs][CERT-SDS]
-* [tls_certificate_certificate_provider][CERT-PROVIDER]
-* [validation_context_sds_secret_config][VAL-SDS]
-* [validation_context_certificate_provider][VAL-PROVIDER]
-
-When fields of type [`CertificateProviderPluginInstance`][CPPI] (or the deprecated
-[`CertificateProviderInstance`][CPI]) are present gRPC ignores the respective
-unsupported fields but if the unsupported fields are present by themselves gRPC will NACK an update
-as part of validating all CDS/LDS updates.
-
-For example, in [`UpstreamTlsContext`][UTC] (on the client side), the identity certificate is not
-needed for the TLS mode, which is indicated by the absence of both [`tls_certificate_provider_instance`][TCPI]
-and [`tls_certificate_certificate_provider_instance`][TCCPI]. However in this case if any of the other
-unsupported identity certificate fields ([tls_certificates][TLS-CERT], [tls_certificate_sds_secret_configs][CERT-SDS],
-[tls_certificate_certificate_provider][CERT-PROVIDER]) are present then gRPC will NACK a CDS update.
-But if [`tls_certificate_provider_instance`][TCPI] (or in its absence the deprecated
-[`tls_certificate_certificate_provider_instance`][TCCPI]) is present, then the unsupported
-fields are ignored. This allows a control plane to formulate a single `CommonTlsContext` that works
-for both gRPC (which only supports [`CertificateProviderPluginInstance`][CPPI] or the deprecated
-[`CertificateProviderInstance`][CPI]) and Envoy (which supports neither [`CertificateProviderPluginInstance`][CPPI]
-nor [`CertificateProviderInstance`][CPI]) to achieve consistent security configuration between
-gRPC and Envoy.
-
-The following fields are unsupported and if present will cause a NACK from gRPC for any update
+* [tls_certificates][TLS-CERT]: if this is present but [`tls_certificate_provider_instance`][TCPI] is
+not present, then gRPC will NACK the update.
+* [tls_certificate_sds_secret_configs][CERT-SDS]: similar to above, if this is present but
+[`tls_certificate_provider_instance`][TCPI] is not present, then gRPC will NACK the update.
+* [validation_context_sds_secret_config][VAL-SDS]: if this is present but a
```

Comment:
This is true only on the server side.  On the client side, if there is no `CertificateValidationContext` from either `validation_context` or `combined_validation_context.default_validation_context`, then we NACK the CDS update without ever looking at the `validation_context_sds_secret_config` field.

I think we should remove this bullet and instead note this in the server-side section above.  I suggest rewriting the paragraph starting on line 291 to read as follows:

"""
If a secure server is configured for mTLS, it will need configuration for how to validate the client's certificate. This validation is configured via the `CertificateValidationContext` message, which comes from one of the options in the `validation_context_type` oneof.  If the `validation_context_sds_secret_config` field is set, gRPC will NACK the CDS update, since we do not support SDS.  If the [`validation_context`][validation_context] field is set, we get the `CertificateValidationContext` from there.  If the [`combined_validation_context`][CVC] field is set, we get the `CertificateValidationContext` from its [`default_validation_context`][default_validation_context] field.  If there is no `CertificateValidationContext`, then the server will not request the client's certificate during the TLS handshake (i.e., it will use TLS instead of mTLS).
"""

Reply:
done